internal/cmd/perfcomp/main.go

```diff
@@ -0,0 +1,357 @@
+	if err != nil {
+		log.Fatalf("Error getting raw data: %v", err)
+	}
+
+	allEnergyStats, err := getEnergyStatsForAllBenchMarks(patchRawData, db.Collection(stableRegionsColl))
+	if err != nil {
+		log.Fatalf("Error getting energy statistics: %v", err)
+	}
+	log.Println(generatePRComment(allEnergyStats, version))
+
+	// Disconnect client
+	err = client.Disconnect(context.Background())
+	if err != nil {
+		log.Fatalf("Failed to disconnect client: %v", err)
+	}
+}
+
+func findRawData(version string, coll *mongo.Collection) ([]RawData, error) {
+	filter := bson.D{
+		{"info.project", "mongo-go-driver"},
+		{"info.version", version},
+		{"info.variant", "perf"},
+		{"info.task_name", "perf"},
+	}
+
+	findCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	cursor, err := coll.Find(findCtx, filter)
+	if err != nil {
+		log.Fatalf(
+			"Error retrieving raw data for version %q: %v",
+			version,
+			err,
+		)
+	}
+	defer func() { err = cursor.Close(findCtx) }()
+
+	log.Printf("Successfully retrieved %d docs from version %s.\n", cursor.RemainingBatchLength(), version)
+
+	var rawData []RawData
+	err = cursor.All(findCtx, &rawData)
+	if err != nil {
+		log.Fatalf(
+			"Error decoding raw data from version %q: %v",
+			version,
+			err,
+		)
+	}
+
+	return rawData, err
+}
+
+// Find the most recent stable region of the mainline version for a specific test/measurement
+func findLastStableRegion(testname string, measurement string, coll *mongo.Collection) (*StableRegion, error) {
+	filter := bson.D{
+		{"time_series_info.project", "mongo-go-driver"},
+		{"time_series_info.variant", "perf"},
+		{"time_series_info.task", "perf"},
+		{"time_series_info.test", testname},
+		{"time_series_info.measurement", measurement},
+		{"last", true},
+		{"contexts", bson.D{{"$in", bson.A{"GoDriver perf task"}}}},
+	}
+
+	findOptions := options.FindOne().SetSort(bson.D{{"end", -1}})
+
+	findCtx, cancel := context.WithTimeout(context.Background(), 180*time.Second)
+	defer cancel()
+
+	var sr *StableRegion
+	err := coll.FindOne(findCtx, filter, findOptions).Decode(&sr)
+	if err != nil {
+		return nil, err
+	}
+	return sr, nil
+}
+
+// For a specific test and measurement
+func getEnergyStatsForOneBenchmark(rd RawData, coll *mongo.Collection) ([]*EnergyStats, error) {
+	testname := rd.Info.TestName
+	var energyStats []*EnergyStats
+
+	for i := range rd.Rollups.Stats {
+		measurement := rd.Rollups.Stats[i].Name
+		patchVal := []float64{rd.Rollups.Stats[i].Val}
+
+		stableRegion, err := findLastStableRegion(testname, measurement, coll)
+		if err != nil {
+			log.Fatalf(
+				"Error finding last stable region for test %q, measurement %q: %v",
+				testname,
+				measurement,
+				err,
+			)
+		}
+
+		pChange := getPercentageChange(patchVal[0], stableRegion.Mean)
+		e, t, h := getEnergyStatistics(mat.NewDense(len(stableRegion.Values), 1, stableRegion.Values), mat.NewDense(1, 1, patchVal))
```

Comment:
Suggest pre-defining the arguments to `getEnergyStatistics` to make the input more clear:

```go
stabilityRegionVec := mat.NewDense(len(stableRegion.Values), 1, stableRegion.Values)
measSingleton := mat.NewDense(1, 1, patchVal)

estat, tstat, hscore := getEnergyStatistics(stabilityRegionVec, measSingleton)
```